### PR TITLE
PostProcess: Fixed WebGL warnings (close #190)

### LINKED ISSUE #190: Post-processing effects generate compiler warnings in WebGL
Also, bloom uses `TEXTURE_ADDRESS_BORDER` that is not supported on the Web.

```
Tutorial27_PostProcessing.html:47 WARNING: 0:1773: '[]' : Performance: dynamic indexing of vectors and matrices is emulated and can be slow.
Tutorial27_PostProcessing.html:47 WARNING: 0:1774: '[]' : Performance: dynamic indexing of vectors and matrices is emulated and can be slow.
Tutorial27_PostProcessing.html:47 WARNING: 0:1775: '[]' : Performance: dynamic indexing of vectors and matrices is emulated and can be slow.
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: Info: Compiler output for shader 'ComputeStencilMaskAndExtractRoughnessPS':
Tutorial27_PostProcessing.html:47 WARNING: 0:1818: '[]' : Performance: dynamic indexing of vectors and matrices is emulated and can be slow.
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: Info: Compiler output for shader 'ComputeTemporalAccumulationPS':
Tutorial27_PostProcessing.html:47 WARNING: 0:1933: 'warning' : unrecognized pragma
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: Info: Compiler output for shader 'ComputePrefilteredDepthBufferPS':
Tutorial27_PostProcessing.html:47 WARNING: 0:1903: 'warning' : unrecognized pragma
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: Info: Compiler output for shader 'ComputeAmbientOcclusionPS':
Tutorial27_PostProcessing.html:47 WARNING: 0:1940: '[]' : Performance: dynamic indexing of vectors and matrices is emulated and can be slow.
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: Info: Compiler output for shader 'ComputeTemporalAccumulationPS':
Tutorial27_PostProcessing.html:47 WARNING: 0:1951: 'warning' : unrecognized pragma
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: Info: Compiler output for shader 'ComputeResampledHistoryPS':
Tutorial27_PostProcessing.html:47 WARNING: 0:1951: 'warning' : unrecognized pragma
Tutorial27_PostProcessing.html:47 WARNING: 0:2062: '[]' : Performance: dynamic indexing of vectors and matrices is emulated and can be slow.
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: Info: Compiler output for shader 'ComputeTemporalAccumulationPS':
Tutorial27_PostProcessing.html:47 WARNING: 0:1840: 'warning' : unrecognized pragma
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: ERROR: GL_CLAMP_TO_BORDER filtering mode is not supported. Defaulting to GL_CLAMP_TO_EDGE.
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: ERROR: GL_CLAMP_TO_BORDER filtering mode is not supported. Defaulting to GL_CLAMP_TO_EDGE.
Tutorial27_PostProcessing.html:47 
Tutorial27_PostProcessing.html:47 Diligent Engine: ERROR: GL_CLAMP_TO_BORDER filtering mode is not supp
```

### PR DESCRIPTION
**Pull request by MikhailGorobets**

### CODE REVIEW

File: PostProcess/Bloom/src/Bloom.cpp [hunk 1]
```diff
@@ -152,6 +152,8 @@ Int32 Bloom::ComputeMipCount(Uint32 Width, Uint32 Height, float Radius)
 
 void Bloom::ComputePrefilteredTexture(const RenderAttributes& RenderAttribs)
 {
+    auto const& SupportedFeatures = RenderAttribs.pPostFXContext->GetSupportedFeatures();
```
Comment (TheMostDiligent) [inline PostProcess/Bloom/src/Bloom.cpp:155]: This is only specific to bloom, no need to have this in FX context

File: PostProcess/Bloom/src/Bloom.cpp [hunk 2]
```diff
@@ -162,7 +164,7 @@ void Bloom::ComputePrefilteredTexture(const RenderAttributes& RenderAttribs)
         ResourceLayout
             .AddVariable(SHADER_TYPE_PIXEL, "cbBloomAttribs", SHADER_RESOURCE_VARIABLE_TYPE_STATIC)
             .AddVariable(SHADER_TYPE_PIXEL, "g_TextureInput", SHADER_RESOURCE_VARIABLE_TYPE_DYNAMIC)
-            .AddImmutableSampler(SHADER_TYPE_PIXEL, "g_TextureInput", Sam_LinearBoarder);
+            .AddImmutableSampler(SHADER_TYPE_PIXEL, "g_TextureInput", SupportedFeatures.BorderSamplingModeSupported ? Sam_LinearBoarder : Sam_LinearClamp);
```
Comment (TheMostDiligent) [inline PostProcess/Bloom/src/Bloom.cpp:167]: Simply query this from device directly

File: Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputeAmbientOcclusion.fx [hunk 3]
```diff
@@ -38,7 +38,10 @@ float SamplePrefilteredDepth(float2 ScreenCoordUV, float MipLevel)
 
 float2 ComputeSliceDirection(float Xi, int Index)
 {
-    float3 Rotations = float3(0.0, 120.0, 240);
+    float Rotations[3];
+    Rotations[0] = 0.0;
+    Rotations[1] = 120.0;
+    Rotations[2] = 240.0;
     float Rotation = Rotations[Index] / 360.0;
     float Phi = (Xi + Rotation) * M_PI;
```
Comment (TheMostDiligent) [inline Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputeAmbientOcclusion.fx:46]: Why does this need to be so complicated? Why not `float(Index) / 3.0`?

File: Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputePrefilteredDepthBuffer.fx [hunk 4]
```diff
@@ -4,7 +4,9 @@
 #include "FullScreenTriangleVSOutput.fxh"
 #include "PostFX_Common.fxh"
 
-#pragma warning(disable : 3078)
+#if !defined(GLSL) && !defined(GL_ES)
+    #pragma warning(disable : 3078)
```
Comment (TheMostDiligent) [inline Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputePrefilteredDepthBuffer.fx:8]: Why is this pragma needed BTW?

**Review by TheMostDiligent — CHANGES_REQUESTED** [review]

File: PostProcess/Bloom/src/Bloom.cpp (hunk 1 above)
Comment (MikhailGorobets) [reply PostProcess/Bloom/src/Bloom.cpp:155]: Let's keep it for consistency with other post effects. It's not a fact that it won't be needed in the future

File: Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputePrefilteredDepthBuffer.fx (hunk 4 above)
Comment (MikhailGorobets) [reply Shaders/PostProcess/ScreenSpaceAmbientOcclusion/private/SSAO_ComputePrefilteredDepthBuffer.fx:8]: In FXC compiler `for` loop  doesn't work as in C++, iterated variable in global visibility scope

**Comment by TheMostDiligent** [comment]
Reworked this in d0b1d81ddf852c8ee163412538128c49ac93d1e5

**Pull request closed without merging**